WelcomeWindow.java

```diff
@@ -1,181 +1,538 @@
+         // 1, 2, 3, 4, 5 or 6
          try
          {
-            gameSelection = keyboard.nextInt();
-            if (gameSelection >= 0)
-            {
-            pass = true;
-            }else
-            {
-               System.out.println("Input must be greater that O");
-            }//if user inputs 0 or less
-         } catch (InputMismatchException error1)
+            // Reading the users entry from the JOptionPane and storing it in the gameSelection variable.
+            gameSelection = Integer.parseInt(JOptionPane.showInputDialog(null,
+                    "Divoc Gaming supports the games\n" +
+                            "1.) BattleShots 2.0\n" +
+                            "2.) Rachel's game - Description\n" +
+                            "3.) Seong's game - Description \n" +
+                            "4.) Truth or Drink\n" +
+                            "5.) Surprise me!\n" +
+                            "6.) Exit\n" +
+                            "Please select a game by entering the number\n " +
+                            "associated with your chosen game from the menu above.",
+                    "Options Window", JOptionPane.INFORMATION_MESSAGE));//Question for amount of rounds
+            // Declaring an if statement which checks if the entry is one of the menu options and marks it as
+            // valid if it is.
+            if (Arrays.asList(1,2,3,4,5,6).contains(gameSelection)){
+               // Setting the invalid variable to false only if gameSelction in list.
+               invalid = false;
+            };//if
+         }//try
+         catch (Exception entryError)
          {
-            System.out.println("Selection must be a number between 1 and 6");
-            keyboard.nextLine();
-            pass = false;
-         }//If user inputs a letter
-      } while (pass != true);
-   }
+            // Informing the user that the system doesn't accept a non-numeric format at this stage and asking them
+            // to re-enter.
+            JOptionPane.showMessageDialog(null, "ERROR - Entry Error\n"+
+                    "Please enter a number from the options provided.");
+         }//catch
+
+      } while (invalid == true); //do-while: gameSelection
+
+   }// Method: options
+
 
    public static void main(String[] args)
    {
+      // Calling the welcome method to welcome the user to the application.
       welcome();
+      // Calling the gaming options and prompting the user to make their selection.
       options();
 
-      // Setting up a terminator
+      // Setting up a terminator for the following while loop
       String terminator = "play";
 
-      // Declaring a while loop which allows the user to continue playing after completing their initial game
+      // Declaring a while loop which allows the user to play multiple games/rounds back to back
       while (terminator != "exit")
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.